### ISSUE TITLE
[QUESTION]: WiFiSTA - Why is autoConnect Code commented out?

### ISSUE THREAD

**Issue by hasenradball** [espressif/arduino-esp32#3803]
Why ist the code for WiFi.autoConnect commented out?

```
/**
 * Setting the ESP32 station to connect to the AP (which is recorded)
 * automatically or not when powered on. Enable auto-connect by default.
 * @param autoConnect bool
 * @return if saved
 */
bool WiFiSTAClass::setAutoConnect(bool autoConnect)
{
    /*bool ret;
    ret = esp_wifi_set_auto_connect(autoConnect);
    return ret;*/
    return false;//now deprecated
}

/**
 * Checks if ESP32 station mode will connect to AP
 * automatically or not when it is powered on.
 * @return auto connect
 */
bool WiFiSTAClass::getAutoConnect()
{
    /*bool autoConnect;
    esp_wifi_get_auto_connect(&autoConnect);
    return autoConnect;*/
    return false;//now deprecated
}
```

**Comment by lbernstone** [comment]
It has been deprecated in esp-idf

**Comment by hasenradball** [comment]
What is the reason for that?
Does it not work?

**Comment by atanisoft** [comment]
@hasenradball it is enabled by default if you call WiFi.begin() (no args). The esp-idf code doesn't have a public API to expose this or one that is supported.

**Comment by atanisoft** [comment]
You could also look at that commented code as "planned support that never materialized"

**Comment by hasenradball** [comment]
@atanisoft thanks for your answer.
Did I understood you right, that the autoconnect should work by default?
I tried but it doesn‘t with my esp32.

**Comment by atanisoft** [comment]
It depends on how you use the WiFi library. It will only reconnect to the last SSID if persistence is enabled and you call the WiFi.begin() API. Note that if you lose connection without a restart it may not reconnect without your app recalling WiFi.begin().

**Comment by hasenradball** [comment]
Ok understood, so this is a difference to the small brother 8266, there is no need to call WiFi.begin() when connection is lost, or during restart without calling WiFi.begin().

**Comment by lbernstone** [comment]
Autoconnect means the device uses saved credentials to connect immediately at boot.
Reconnect means to retry if a connection is lost.

**Comment by hasenradball** [comment]
Yes so is my understanding! And when the esp32 needs a wifi.begin() to connect, then the esp32 has actually NO autoconnnect and autoreconnect!
But why???

**Comment by hasenradball** [comment]
Why can‘t we do like the 8266?

**Comment by hasenradball** [comment]
A Anwer is left, actaually.

**Comment by ElToberino** [comment]
Hello stale bot... ,
this question is still pending. 
I'd also prefer an auto connect and auto reconnect mode without the need of calling WiFi.begin()
